Translate faq-versioning to spanish (#154)
reactjs/es.reactjs.org: content/docs/faq-versioning.md

---
id: faq-versioning
title: Política de versiones
permalink: docs/faq-versioning.html
layout: docs
category: FAQ
---

React sigue los principios de [versionado semántico (semver)](https://semver.org/lang/es/).

Esto significa que con un número de versión **x.y.z**:

* Al publicar **cambios de ruptura**, hacemos una **publicación mayor** cambiando el número **x** (ej: 15.6.2 a 16.0.0).
* Al publicar **nueva funcionalidad**, hacemos una **publicación menor** cambiando el número **y** (ej: 15.6.2 a 15.7.0).

> * Al publicar **correcciones de bugs**, publicamos un **patch** cambiando el número **z** (ej: 15.6.2 a 15.6.3).

Author: me parece bien 👍

### Mejoras graduales a través de advertencias {#gradual-upgrades-via-warnings}

Las versiones de desarrollo de React incluyen muchas advertencias útiles. Siempre que sea posible, añadimos advertencias en preparación a futuros cambios de ruptura. De esta manera, si tu aplicación no tiene advertencias en la última versión, ella será compatible con la próxima versión mayor. Esto te permite actualizar tu aplicación un componente a la vez.

> Las advertencias de desarrollo no afectarán el comportamiento en tiempo de ejecución de tu aplicación. De esa manera, puedes sentir la confianza de que tu aplicación se comportará de la misma manera entre la versión de desarrollo y la versión de producción -- las únicas diferencias son que la versión de producción no registrará las advertencias y es más eficiente. (Si alguna vez observa lo contrario, por favor presentar un *issue*.)

Reviewer: ```suggestion
Las advertencias de desarrollo no afectarán el comportamiento en tiempo de ejecución de tu aplicación. De esa manera, puedes estar seguro de que tu aplicación se comportará de la misma manera entre la versión de desarrollo y la versión de producción -- las únicas diferencias son que la versión de producción no registrará las advertencias y es más eficiente. (Si alguna vez observas lo contrario, por favor abre un caso).
```

> Las advertencias de desarrollo no afectarán el comportamiento en tiempo de ejecución de tu aplicación. De esa manera, puedes estar seguro de que tu aplicación se comportará de la misma manera entre la versión de desarrollo y la versión de producción -- las únicas diferencias son que la versión de producción no registrará las advertencias y es más eficiente. (Si alguna vez observas lo contrario, por favor abre un caso).

Reviewer: ```suggestion
Las advertencias de desarrollo no afectarán el comportamiento en tiempo de ejecución de tu aplicación. De esa forma, puedes estar seguro de que tu aplicación se comportará de la misma manera entre la versión de desarrollo y la versión de producción -- las únicas diferencias son que la versión de producción no registrará las advertencias y es más eficiente. (Si alguna vez observas lo contrario, por favor abre un caso).
```

> ### ¿Qué cuenta como un cambio de ruptura? {#what-counts-as-a-breaking-change}

Reviewer: ```suggestion
### ¿Qué cuenta como un Cambio con Rupturas? {#what-counts-as-a-breaking-change}
```

> En general, *no publicamos* una versión mayor por cambios a:

Reviewer: ```suggestion
En general, *no publicamos* una versión importante por cambios a:
```

> * **Advertencias de desarrollo.** Dado que no afectan al comportamiento de producción, podemos añadir nuevas advertencias o modificar advertencias existentes entre las versiones mayores. De hecho, esto es lo que nos permite advertir de forma fiable sobre los próximos cambios de ruptura.

Reviewer: ```suggestion
* **Advertencias de desarrollo.** Dado que no afectan el comportamiento en producción, podemos añadir nuevas advertencias o modificar advertencias existentes entre versiones importantes. De hecho, esto es lo que nos permite advertir de forma fiable sobre los próximos cambios con rupturas.
```

> * **APIs que comienzan con `unstable_`.** Estos ofrecen funcionalidades experimentales sobre cuyos APIs todavía no estamos seguros. Al publicar esto con un prefijo `unstable_`, podemos iterar más rápido y llegar a un API estable lo antes posible.

Reviewer: ```suggestion
* **APIs que comienzan con `unstable_`.** Estas ofrecen funcionalidades experimentales sobre cuyos APIs todavía no estamos seguros. Al publicar esto con un prefijo `unstable_`, podemos iterar más rápido y llegar a un API estable lo antes posible.
```

Author: API es masculino o femenino 🤔 ?

Reviewer: Normalmente se utiliza con `la` aunque tambien he dudado en esta, hasta leer este articulo: https://www.rubenbejar.com/2017/11/la-API/ 😅

Author: 😮 buen articulo, ambos me parecen bien voy a aceptar tu recomendación 👍

> * **APIs que comienzan con `unstable_`.** Estas ofrecen funcionalidades experimentales sobre cuyos APIs todavía no estamos seguros. Al publicar esto con un prefijo `unstable_`, podemos iterar más rápido y llegar a un API estable lo antes posible.

Reviewer: ```suggestion
* **APIs que comienzan con `unstable_`.** Estas ofrecen funcionalidades experimentales sobre cuyas APIs todavía no estamos seguros. Al publicar esto con un prefijo `unstable_`, podemos iterar más rápido y llegar a un API estable lo antes posible.
```

> * **Versiones alfa y canarias de React.**  Proporcionamos versiones alfa de React como una manera de probar nuevas características temprano, pero necesitamos la flexibilidad para hacer cambios basados en lo que aprendemos en el período alfa.

Reviewer: Es común ver versiones alfa bajo ese nombre, pero `canary` traducido a `canarias` no tanto. Creo que esto se podría dejar como el original.

```suggestion
* **Versiones alfa y *canary* de React.**  Proporcionamos versiones alfa de React como una manera de probar nuevas características con antelación, pero necesitamos la flexibilidad para hacer cambios basados en lo que aprendemos en el período alfa.
```

Adicionalmente, falta traducir: `If you use these versions, note that APIs may change before the stable release.`

> * **Versiones alfa y *canary* de React.**  Proporcionamos versiones alfa de React como una manera de probar nuevas características con antelación, pero necesitamos la flexibilidad para hacer cambios basados en lo que aprendemos en el período alfa. Si utiliza estas versiones, tenga en cuenta que los APIs pueden cambiar antes de la versión estable sea publicada.

Reviewer: Solo este último cambio 😅 

```suggestion
* **Versiones alfa y *canary* de React.**  Proporcionamos versiones alfa de React como una manera de probar nuevas características con antelación, pero necesitamos la flexibilidad para hacer cambios basados en lo que aprendemos en el período alfa. Si utilizas estas versiones, ten en cuenta que las APIs pueden cambiar antes de la versión estable sea publicada.
```

> * **APIs no documentadas y estructuras de datos internas.** Si accede a nombres de propiedad interna como `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` o `__reactInternalInstance$uk43rzhitjg`, no podemos garantizar nada. Usted está por su cuenta.

Reviewer: ```suggestion
* **APIs no documentadas y estructuras de datos internas.** Si accedes a nombres de propiedades internas como `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` o `__reactInternalInstance$uk43rzhitjg`, no podemos garantizar nada. Estás por tu cuenta.
```

Author: Gracias! Se me paso el tú vs usted.

> Esta política está diseñada para ser pragmática: desde luego, no queremos causarle dolores de cabeza. Si publicaramos una nueva versión mayor por cada uno de estos cambios, acabaríamos publicando más versiones mayores y causaríamos más inconveniencias a la comunidad. También significaría que no podríamos mejorar React tan rápido como nos gustaría.

Reviewer: ```suggestion
Esta política está diseñada para ser pragmática: desde luego, no queremos causarte dolores de cabeza. Si publicáramos una nueva versión importante por cada uno de estos cambios, acabaríamos lanzando más versiones importantes y causaríamos más inconvenientes con el versionado a la comunidad. También significaría que no podríamos mejorar React tan rápido como nos gustaría.
```

> Dicho esto, si creemos que algún cambio en esta lista va a provocar grandes problemas en la comunidad, haremos todo lo posible para proporcionar un camino de migración gradual.

Reviewer: ```suggestion
Dicho esto, si creemos que algún cambio en esta lista va a provocar grandes problemas en la comunidad, haremos todo lo posible para proporcionar una guía de migración gradual.
```

> Dicho esto, si creemos que algún cambio en esta lista va a provocar grandes problemas en la comunidad, haremos todo lo posible para proporcionar una guía de migración gradual.

Reviewer: ```suggestion
Dicho esto, si creemos que algún cambio en esta lista va a provocar grandes problemas en la comunidad, haremos todo lo posible para proporcionar una guía de actualización gradual.
```
